whd/dispatcher.py

```diff
@@ -211,6 +212,57 @@ def _set_pr_labels(self, pr_event, resources):
                     f" in '{repository_path}' found. Ignoring, since they are not an org member'."
                 )
 
+    def _start_not_triggered_jobs(self, concourse_api, resources):
+        for resource in resources:
+            if resource.type != "pull-request":
+                continue
+
+            resource_versions = concourse_api.resource_versions(
+                resource.pipeline.name,
+                resource.name
+            )
+            if not resource_versions:
+                continue
+
+            jobs_with_triggering_resource = filter(
+                lambda job: job.has_triggering_resource_in_get_step(resource.name),
+                resource.pipeline.jobs
+            )
+            for job in jobs_with_triggering_resource:
+                builds = concourse_api.job_builds(job.pipeline.name, job.name)
+                for resource_version in self._resource_version_ids_not_build(
+                    resource_versions, builds
+                ):
+                    logger.info(
+                        f"Pin resource version {resource_version.version()} "
+                        f"and trigger build for job {job.name}"
+                    )
+                    self._pin_trigger_unpin(job, resource.name, resource_version.id())
+
+    def _resource_version_ids_not_build(self, resource_versions, builds):
+        for resource_version in resource_versions[:10]:
+            resource_version_in_build = False
+            for build in builds[-10:]:
```

Comment:
rather process from most recent to less recent; also: add a comment stating _why_ we only consider the last couple of build executions (also: rather more than 10; maybe as much as 50, even? I do not think this will cause a performance issue - as opposed to iterating over _all_ build executions, of course)